src/Fixer/Phpdoc/PhpdocVarWithoutNameFixer.php

```diff
@@ -68,21 +69,30 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens)
                 continue;
             }
 
-            $doc = new DocBlock($token->getContent());
+            $nextIndex = $tokens->getNextMeaningfulToken($index);
+            $nextToken = isset($tokens[$nextIndex]) ? $tokens[$nextIndex] : null;
 
-            // don't process single line docblocks
-            if (1 === \count($doc->getLines())) {
-                continue;
+            // For people writing static public $foo instead of public static $foo
+            if (null !== $nextToken && $nextToken->isGivenKind(T_STATIC)) {
+                $nextIndex = $tokens->getNextMeaningfulToken($nextIndex);
+                $nextToken = isset($tokens[$nextIndex]) ? $tokens[$nextIndex] : null;
             }
 
-            $annotations = $doc->getAnnotationsOfType(['param', 'return', 'type', 'var']);
-
-            // only process docblocks where the first meaningful annotation is @type or @var
-            if (!isset($annotations[0]) || !\in_array($annotations[0]->getTag()->getName(), ['type', 'var'], true)) {
+            // We want only doc blocks that are for properties and thus have specified access modifiers next
+            if (null === $nextToken || !$nextToken->isGivenKind([T_PRIVATE, T_PROTECTED, T_PUBLIC, T_VAR])) {
```

Comment:
Maybe you can utilize `TokensAnalyzer::getClassyElements` to prevent have this kind of logic duplicated in the code base. Please add a unit test for nested anonymous classes as well.

Reply:
- [X] Nested anonymous classes testcase
- [ ] Try `TokensAnalyzer::getClassyElements`

Reply:
I've looked into `TokensAnalyzer::getClassyElements` however I don't see it becoming cleaner code:

- If working forward from the doc block (like now) we still need a way to match it to the classy element it belongs to.
- If working backwards from the classy elements, we still need to traverse back to find the actual matching doc block if any (and not any intermediary comment that happens to be like `/** this */`) which can be any number of tokens away from the index of the classy element.